lapixdl/formats/annotation.py

```diff
@@ -0,0 +1,204 @@
+        """Tuple[Tuple[int, int], ...]: A tuple with each point of the Bounding Box."""
+        return (
+            self.upper_left_point,
+            self.upper_right_point,
+            self.bottom_right_point,
+            self.bottom_left_point,
+        )
+
+    @property
+    def xy(self) -> tuple[list[int], list[int]]:
+        """Tuple[list[int], list[int]]: A tuple with lists of coords for each x and y"""
+        _x, _y = zip(*self.coords)
+        return (list(_x), list(_y))
+
+    @property
+    def slice_x(self) -> slice:
+        return slice(self.upper_left_x, self.upper_right_point[0])
+
+    @property
+    def slice_y(self) -> slice:
+        return slice(self.upper_left_y, self.bottom_left_point[1])
+
+    def intersection_area_with(self: BBox, bbox: BBox) -> int:
+        """Calculates the intersection area with another bbox
+
+        Args:
+            self (BBox): This bbox
+            bbox (BBox): Bbox to intersect with
+
+        Returns:
+            int: The intersection area with the bbox
+        """
+
+        # Gets each box upper left and bottom right coordinates
+        (upr_lft_x_a, upr_lft_y_a) = self.upper_left_point
+        (btm_rgt_x_a, btm_rgt_y_a) = self.bottom_right_point
+
+        (upr_lft_x_b, upr_lft_y_b) = bbox.upper_left_point
+        (btm_rgt_x_b, btm_rgt_y_b) = bbox.bottom_right_point
+
+        # Calculates the intersection box upper left and bottom right coordinates
+        (upr_lft_x_intersect, upr_lft_y_intersect) = (
+            max(upr_lft_x_a, upr_lft_x_b), max(upr_lft_y_a, upr_lft_y_b))
+        (btm_rgt_x_intersect, btm_rgt_y_intersect) = (
+            min(btm_rgt_x_a, btm_rgt_x_b), min(btm_rgt_y_a, btm_rgt_y_b))
+
+        # Calculates the height and width of the intersection box
+        (w_intersect, h_intersect) = (btm_rgt_x_intersect - upr_lft_x_intersect + 1,
+                                      btm_rgt_y_intersect - upr_lft_y_intersect + 1)
+
+        # If H or W <= 0, there is no intersection
+        if (w_intersect <= 0) or (h_intersect <= 0):
+            return 0
+
+        return w_intersect * h_intersect
+
+    def union_area_with(
+            self: BBox,
+            bbox: BBox,
+            intersection_area: int | None = None
+    ) -> int:
+        """Calculates the union area with another bbox
+
+        Args:
+            self (BBox): This bbox
+            bbox (BBox): Bbox to union with
+            intersection_area (Optional[int], optional): The intersection area between this and bbox. Defaults to None.
+
+        Returns:
+            int: The union area with the bbox
+        """
+
+        return self.area + bbox.area - (intersection_area or self.intersection_area_with(bbox))
+
+    def to_polygon(self) -> Polygon:
+        return Polygon(self.coords)
+
+
+@dataclass
+class Annotation:
+    geometry: Polygon | MultiPolygon
+    category_id: int
+    iscrowd: int = 0
+
+    @property
+    def bbox(self) -> BBox:
```

Comment:
```suggestion
    def to_bbox(self) -> BBox:
```

Reply:
I add this class, but, isn't used in anywhere now, so I will just remove it. 

It is added with an idea of interface when unpacking annotations from the data frame... when this be usefully I add this back